src-built-in/components/advancedAppCatalog/src/stores/storeActions.js

```diff
@@ -74,98 +85,67 @@ function getToolbarStore(done) {
 	});
 }
 
+/**
+ * Return to the App Catalog home page
+ */
+function goHome() {
+	clearFilteredApps();
+	clearApp();
+	clearTags();
+	clearSearchText();
+	setForceSearch(false);
+}
+
 /**
  * Private function to add an active tag. This will filter apps based on tags
- * NOTE: This will need to use search
  * @param {string} tag The name of the tag
  */
 function _addActiveTag(tag) {
-
-	let { apps, activeTags } = data;
-
+	let activeTags = data.activeTags;
 	if (!activeTags.includes(tag)) {activeTags.push(tag);}
 
-	let newApps = apps.filter((app) => {
-		for (let i = 0; i < activeTags.length; i++) {
-			let tag = activeTags[i].trim();
-			if (app.tags.includes(tag)) {
-				return true;
-			}
-		}
-	});
-
-	getStore().setValues([
-		{
-			field: "filteredApps",
-			value: newApps
-		},
-		{
-			field: "activeTags",
-			value: activeTags
-		}
-	]);
+	getStore().setValue({ field: "activeTags", value: activeTags }, filterApps());
 }
 
 /**
  * Private function to remove an active tag. This will filter apps based on tags
- * NOTE: This will need to use search
  * @param {string} tag The name of the tag
  */
 function _removeActiveTag(tag) {
-
-	let { activeTags, apps } = data;
-
-	let newActiveTags = activeTags.filter((currentTag) => {
+	data.activeTags = data.activeTags.filter((currentTag) => {
 		return currentTag !== tag;
 	});
 
-	let newApps = apps.filter((app) => {
-		for (let i = 0; i < newActiveTags.length; i++) {
-			let tag = activeTags[i].trim();
-			if (app.tags.includes(tag)) {
-				return true;
-			}
-		}
-	});
-
-	getStore().setValues([{
-		field: "activeTags",
-		value: newActiveTags
-	}, {
-		field: "filteredApps",
-		value: newApps
-	}]);
-}
-
-function _refreshTags() {
-	let { activeTags, apps } = data;
-
-	let newApps = apps.filter((app) => {
-		for (let i = 0; i < activeTags.length; i++) {
-			const tag = activeTags[i].trim();
-			if (app.tags.includes(tag)) {
-				return true;
-			}
-		}
-	});
-
-	getStore().setValues([{
-		field: 'activeTags',
-		value: activeTags
-	}, {
-		field: 'filteredApps',
-		value: newApps
-	}]);
+	getStore().setValue({ field: "activeTags", value: data.activeTags	});
+	filterApps();
 }
 
 /**
  * Clears all active tags
  */
 function _clearActiveTags() {
-	getStore().setValue({
-		field: "activeTags",
-		value: []
-	});
+	getStore().setValue({ field: "activeTags", value: [] });	
+}
+
+
+/**
+ * Send the search text and tags to the appd server and get a list of apps
+ */
+function filterApps() {
+	let { activeTags, searchText } = data;
+	
+
+	if (activeTags.length === 0 && searchText === "") {
+		goHome();
+	} else {
+		appd.search({ text: searchText, tags: activeTags }, (err, data) => {
+			if (err) {
+				Logger.system.error("FDC3 App search failed!: ${err}");
```

Comment:
(Clippy Voice): It looks like you're trying to interpolate a variable into a string. In JavaScript this should be done with back-ticks and not double quotes.

Reply:
Oops! Good catch. This has been resolved.